crates/red_knot_python_semantic/src/semantic_index/re_exports.rs

```diff
@@ -3,6 +3,18 @@
 //!
 //! For example, if a module `foo` contains `from bar import *`, which symbols from the global
 //! scope of `bar` are imported into the global namespace of `foo`?
+//!
+//! ## Why is this a separate query rather than a part of semantic indexing?
+//!
+//! This query is called by the [`super::SemanticIndexBuilder`] in order to add the correct
+//! [`super::Definition`]s to the semantic index of a module `foo` if `foo` has a
+//! `from bar import *` statement in its global namespace. Adding the correct `Definition`s to
+//! `foo`'s [`super::SemanticIndex`] requires knowing which symbols are exported from `bar`.
+//!
+//! If we determined the set of exported names during semantic indexing rather than as a
+//! separate query, we would need to complete semantic indexing on `bar` in order to
+//! complete analysis of the global namespace of `foo`. Since semantic indexing is somewhat
```

Comment:
Not sure if we have to say it here, but a main concern is also making semantic indexing potentially cyclic, meaning we'd have to fixpoint iterate it, which would be even more expensive.

Reply:
https://github.com/astral-sh/ruff/pull/16958